cyral/model_integration_idp_generic_saml.go

```diff
@@ -62,6 +73,30 @@ func (uatt *RequiredUserAttributes) WriteToSchema(d *schema.ResourceData) error
 	return d.Set("attributes", attributes)
 }
 
+func (spmetadataObj *GenericSAMLSPMetadata) WriteToSchema(d *schema.ResourceData) error {
+	var spMetadata []interface{}
+	if spmetadataObj != nil {
```

Comment:
If we extract the logic for transforming the SP metadata object into a map to a function (E.g.: `ToMap`) we can reuse this code in the data source as well and get rid of this duplicated code.
```Go
if spMetadataObj != nil {
  spMetadata = append(spMetadata, spMetadataObj.ToMap())
}
```

Reply:
Addr [7b9f350](https://github.com/cyralinc/terraform-provider-cyral/pull/394/commits/7b9f3503aa1721a658226ab4e45695663018b2ba)